test/Grpc.IntegrationTests/AutochunkTests.cs

```diff
@@ -0,0 +1,165 @@
+        });
+
+        string instanceId = await server.Client.ScheduleNewOrchestrationInstanceAsync(orchestratorName);
+        using CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+        OrchestrationMetadata metadata = await server.Client.WaitForInstanceCompletionAsync(
+            instanceId, getInputsAndOutputs: true, cts.Token);
+
+        Assert.NotNull(metadata);
+        Assert.Equal(instanceId, metadata.InstanceId);
+        Assert.Equal(OrchestrationRuntimeStatus.Completed, metadata.RuntimeStatus);
+        Assert.Equal(ActivityCount + TimerCount + SubOrchCount, metadata.ReadOutputAs<int>());
+    }
+
+    /// <summary>
+    /// Validates that when a single orchestrator action exceeds the MaxCompleteOrchestrationWorkItemSizePerChunk limit,
+    /// the orchestration completes with a failed status and proper failure details.
+    /// </summary>
+    [Fact]
+    public async Task Autochunk_SingleActionExceedsChunkSize_CompletesWithFailedStatus()
+    {
+        const int ChunkSize = GrpcDurableTaskWorkerOptions.MinCompleteOrchestrationWorkItemSizePerChunkBytes; // 1 MB
+        // Create a payload that exceeds the chunk size (1 MB + some overhead)
+        const int PayloadSize = ChunkSize + 100 * 1024; // 1.1 MB payload
+        TaskName orchestratorName = nameof(Autochunk_SingleActionExceedsChunkSize_CompletesWithFailedStatus);
+        TaskName activityName = "Echo";
+
+        await using HostTestLifetime server = await this.StartWorkerAsync(b =>
+        {
+            b.UseGrpc(opt => opt.MaxCompleteOrchestrationWorkItemSizePerChunk = ChunkSize);
+            b.AddTasks(tasks => tasks
+                .AddOrchestratorFunc(orchestratorName, async ctx =>
+                {
+                    // Attempt to schedule an activity with a payload that exceeds the chunk size
+                    string largePayload = new string('A', PayloadSize);
+                    return await ctx.CallActivityAsync<string>(activityName, largePayload);
+                })
+                .AddActivityFunc<string, string>(activityName, (ctx, input) => Task.FromResult(input)));
+        });
+
+        string instanceId = await server.Client.ScheduleNewOrchestrationInstanceAsync(orchestratorName);
+        using CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+        OrchestrationMetadata metadata = await server.Client.WaitForInstanceCompletionAsync(
+            instanceId, getInputsAndOutputs: true, cts.Token);
+
+        Assert.NotNull(metadata);
+        Assert.Equal(instanceId, metadata.InstanceId);
+        Assert.Equal(OrchestrationRuntimeStatus.Failed, metadata.RuntimeStatus);
+        Assert.NotNull(metadata.FailureDetails);
+        Assert.Contains("exceeds the", metadata.FailureDetails.ErrorMessage, StringComparison.OrdinalIgnoreCase);
+        Assert.Contains("MB limit", metadata.FailureDetails.ErrorMessage, StringComparison.OrdinalIgnoreCase);
```

Comment:
Can you also check that it contains a reference to the `maxChunkBytes` and that non-retriable is true??